iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2769,7 +2765,7 @@ static int UpdateMqttConnectionStateIfNeeded(PMQTTTRANSPORT_HANDLE_DATA transpor
             }
             else if (retry_action == RETRY_ACTION_STOP_RETRYING)
             {
-                // Set callback if retry expired
+                // send callback if retry expired
                 if (!transport_data->isRetryExpiredCallbackCalled)
```

Comment:
if the above is true does this need to be protected anymore? is this only called in one place?

Reply:
Yes it does. When you queue a new message this event will keep firing for every new message. I initially removed it, but broke the unit test. I added it back. This was previous code, not related to the new event that was added above.